sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -327,7 +327,6 @@ case class InSubquery(values: Seq[Expression], query: ListQuery)
 
   override def children: Seq[Expression] = values :+ query
   override def nullable: Boolean = children.exists(_.nullable)
-  override def foldable: Boolean = children.forall(_.foldable)
```

Comment:
Let's have some more confirmation of why it was written this way originally, and whether or not we're losing anything by just making `folable` always `false` here.

Reply:
seems there was no specific reason: https://github.com/apache/spark/pull/21403 . We just followed `In` at that time.

It's obviously not foldable as it needs to evaluate a subquery.

Reply:
OK +1 with @cloud-fan 's comment.